src/components/Property/Types/VectorProperty/VectorProperty.tsx

```diff
@@ -1,36 +1,101 @@
 
-import { ColorVector } from './ViewOptions/ColorVector';
-import { ValueList } from './ViewOptions/DefaultValueList';
+import { useGetPropertyDescription, useProperty } from '@/api/hooks';
+import { ColorPicker } from '@/components/ColorPicker/ColorPicker';
+import { AdditionalDataVectorMatrix, PropertyProps } from '@/components/Property/types';
+
+import { DefaultView } from './ViewOptions/DefaultView';
 import { MinMaxRange } from './ViewOptions/MinMaxRange';
 
-export interface VectorPropertyProps extends ConcretePropertyBaseProps {
-  setPropertyValue: (newValue: number[]) => void;
-  value: number[];
-  additionalData: {
-    Exponent: number; // TODO: handle the exponent
-    MaximumValue: number[];
-    MinimumValue: number[];
-    SteppingValue: number[];
-  };
-  viewOptions: {
-    Color?: boolean;
-    MinMaxRange?: boolean;
-  };
-  // Not part of the data from OpenSpace
+const vectorPropertyTypes = [
+  'Vec2Property',
+  'Vec3Property',
+  'Vec4Property',
+  'DVec2Property',
+  'DVec3Property',
+  'DVec4Property',
+  'IVec2Property',
+  'IVec3Property',
+  'IVec4Property',
+  'UVec2Property',
+  'UVec3Property',
+  'UVec4Property'
+];
+
+interface Props extends PropertyProps {
   isInt?: boolean;
 }
 
-export function VectorProperty(props: VectorPropertyProps) {
-  const { viewOptions } = props;
+export function VectorProperty({ uri, isInt }: Props) {
+  const [value, setPropertyValue] = useProperty<number[]>(uri, vectorPropertyTypes);
+  const description = useGetPropertyDescription(uri);
+
+  if (!description || !value) {
+    return <></>;
+  }
+
+  const viewOptions = description.metaData.ViewOptions;
+  const additionalData = description.additionalData as AdditionalDataVectorMatrix;
+  const { isReadOnly } = description.metaData;
 
   if (viewOptions.Color) {
-    return <ColorVector {...props} />;
+    if ((value.length !== 3 && value.length !== 4) || isInt) {
+      throw Error('Invalid use of Color view option!');
+    }
+    const hasAlpha = value.length === 4;
+
+    return (
+      <Flex gap={'xs'} align={'center'}>
+        <DefaultView
+          disabled={isReadOnly}
+          setPropertyValue={setPropertyValue}
+          value={value}
+          additionalData={additionalData}
+        />
+        <ColorPicker
+          disabled={isReadOnly}
+          withAlpha={hasAlpha}
+          color={{
+            r: Math.round(255 * value[0]),
+            g: Math.round(255 * value[1]),
+            b: Math.round(255 * value[2]),
+            a: hasAlpha ? value[3] : 1.0
+          }}
+          onChange={(rgbaColor: RGBA) => {
+            const newValue = [rgbaColor.r / 255, rgbaColor.g / 255, rgbaColor.b / 255];
+            if (hasAlpha) {
+              newValue.push(rgbaColor.a);
+            }
+            // @TODO (emmbr26, 2025-03-14) Check for min max values. color values should be
+            // between 0 and 1.
+            setPropertyValue(newValue);
+          }}
+        />
+      </Flex>
```

Comment:
Why was the component for this removed? To me it seems much more clear to have each view option in a separate component

Now it's a bit confusing that there is a component in the viewoptions folder for MinMaxRange, but not for color